File: frontend/src/app/model/BookDetailsResponse.ts
export interface BookDetailsResponse {
  bookId: number,
  title: string,
  price: number,
  originalPrice: number,
  quantity: number,
  publisher: PublisherDetailsResponse,
  authors: Array<AuthorDetailsResponse>,
  numberOfPages: number,
  edition: number,
  numberOfStars: number,
  publicationYear: number,
  description: string,
  category: string,
  subCategory: string,
  pictures: Array<string>,
  cover: string
}

export interface AuthorDetailsResponse {
  authorId: number,
  name: string,
  description: string,
  picture: string
}

export interface PublisherDetailsResponse {
  publisherId: number,
  name: string
}

File: backend/book/src/main/java/pl/tiguarces/controller/response/BookDetailsResponse.java
package pl.tiguarces.controller.response;

import pl.tiguarces.model.Publisher;

import java.util.List;

public record BookDetailsResponse(long bookId, String title, double price, Double originalPrice,
                                  int quantity, Publisher publisher, List<Author> authors,
                                  int numberOfPages, int edition, int numberOfStars,
                                  Integer publicationYear, String description,
                                  String category, String subCategory, List<String> pictures, String cover) {

    public record Author(long authorId, String name, String description, String picture) { }
}
